docs/tutorial.rst

```diff
@@ -0,0 +1,912 @@
+  >>> print(hp.interpolate_bilinear_skycoord(coord, sm.stokes.value[0,0,:]))
+  6540.375582405899
+
+  >>> a = (sm.gl.radian/math.pi)[:3]
+  >>> z = np.cos(sm.gb.radian)[:3]
+  >>> fig, ax = plt.subplots() # doctest: +SKIP
+  >>> ax.scatter(a, z, alpha = 0) # doctest: +SKIP
+  >>> ax.set_xlabel("phi / pi") # doctest: +SKIP
+  >>> ax.set_ylabel("z = cos(theta)") # doctest: +SKIP
+  >>> for i, txt in enumerate(sm.hpx_inds[:3]): # doctest: +SKIP
+  ...     #adds pixel index at center of each pixel
+  ...     ax.annotate(txt, (a[i], z[i]), fontsize=8) # doctest: +SKIP
+  >>> for j in range(len(sm.hpx_inds[:3])): # doctest: +SKIP
+  ...     lon = hp.boundaries_lonlat(sm.hpx_inds[j], 100)[0]/math.pi # doctest: +SKIP
+  ...     lat = np.cos(hp.boundaries_lonlat(sm.hpx_inds[j], 100)[1]) # doctest: +SKIP
+  ...     lon = lon.value # doctest: +SKIP
+  ...     lat = lat.value # doctest: +SKIP
+  ...     vertices = np.vstack([lon.ravel(), lat.ravel()]).transpose() # doctest: +SKIP
+  ...     p = Polygon(vertices, closed=True, edgecolor="black", facecolor="none") # doctest: +SKIP
+  ...     # adds boundaries around each pixel
+  ...     ax.add_patch(p) # doctest: +SKIP
+
+  >>> a = (sm.gl.radian/math.pi)[:3]
+  >>> z = np.cos(sm.gb.radian)[:3]
+  >>> fig, ax = plt.subplots() # doctest: +SKIP
+  >>> ax.scatter(a, z, alpha = 0) # doctest: +SKIP
+  >>> ax.set_xlabel("phi / pi") # doctest: +SKIP
+  >>> ax.set_ylabel("z = cos(theta)") # doctest: +SKIP
+  >>> # nested instead of ring
+  >>> for i, txt in enumerate(hp.ring_to_nested(sm.hpx_inds)[:3]): # doctest: +SKIP
+  ...     ax.annotate(txt, (a[i], z[i]), fontsize=8) # doctest: +SKIP
+  >>> for j in range(len(sm.hpx_inds[:3])): # doctest: +SKIP
+  ...     lon = hp.boundaries_lonlat(sm.hpx_inds[j], 100)[0]/math.pi # doctest: +SKIP
+  ...     lat = np.cos(hp.boundaries_lonlat(sm.hpx_inds[j], 100)[1]) # doctest: +SKIP
+  ...     lon = lon.value # doctest: +SKIP
+  ...     lat = lat.value # doctest: +SKIP
+  ...     vertices = np.vstack([lon.ravel(), lat.ravel()]).transpose() # doctest: +SKIP
+  ...     p = Polygon(vertices, closed=True, edgecolor="black", facecolor="none") # doctest: +SKIP
+  ...     ax.add_patch(p) # doctest: +SKIP
+
+SkyModel: Creating and writing out catalogs
+-------------------------------------------
+
+a) skyh5
+********
+.. code-block:: python
+
+  >>> import os
+  >>> import numpy as np
+  >>> import matplotlib.pyplot as plt # doctest: +SKIP
+  >>> from pyradiosky import SkyModel
+
+  >>> sm = SkyModel(component_type="healpix", nside=1, hpx_inds=[0,1,2,3], stokes=np.zeros((4,1,4)), spectral_type="flat",
+  ...               hpx_order="ring")
+  >>> print(sm.get_lon_lat())
+  (<Longitude [ 45., 135., 225., 315.] deg>, <Latitude [41.8103149, 41.8103149, 41.8103149, 41.8103149] deg>)
+
+  >>> sm.filename == "zero.skyh5"
+  >>> write_file = os.path.join(".", )
+  >>> sm.write_skyh5("zero.skyh5")
+
+b) working with time and location
```

Comment:
Let's add something about the `coherency_calc` method to this section -- that's one of the fancier things we do with time & location information. The source will need to be polarized to really show it off.